src/zenml/cli/stack_components.py

```diff
@@ -1351,12 +1362,165 @@ def flavor_group() -> None:
+                    f"dataset list` to list all available datasets."
+                )
+            unlabeled_task_count = len(project.get_unlabeled_tasks())
+            labeled_task_count = len(project.get_labeled_tasks())
+            total_task_count = unlabeled_task_count + labeled_task_count
+            cli_utils.declare(
+                f"Annotation stats for '{name}' dataset:", bold=True
+            )
+            cli_utils.declare(f"Total annotation tasks: {total_task_count}")
+            cli_utils.declare(f"Labeled annotation tasks: {labeled_task_count}")
+            cli_utils.declare(
+                f"Unlabeled annotation tasks: {unlabeled_task_count}"
+            )
+
+        @dataset.command(
+            "delete", context_settings={"ignore_unknown_options": True}
+        )
+        @click.argument(
+            "name",
+            type=click.STRING,
+        )
+        @click.option(
+            "--all",
+            "-a",
+            "all",
+            is_flag=True,
+            help="Use this flag to delete all datasets.",
+            type=click.BOOL,
+        )
+        @click.pass_obj
+        def dataset_delete(
+            annotator: "BaseAnnotator", name: str, all: bool
+        ) -> None:
+            """Delete a dataset.
+
+            If the --all flag is used, all datasets will be deleted.
+
+            Args:
+                annotator: The annotator stack component.
+                name: Name of the dataset to delete.
+                all: Whether to delete all datasets.
+            """
+            cli_utils.error(
+                "Currently unable to delete datasets. Awaiting updated version from Label Studio."
+            )
+            # TODO: Add this back when Label Studio release a new version
+            # cli_utils.declare(f"Deleting your dataset '{name}'")
+            # if all:
+            #     datasets = annotator.get_datasets()
+            #     for dataset in datasets:
+            #         dataset_name = dataset.get_params()["title"]
+            #         annotator.delete_dataset(dataset_name=dataset_name)
+            # else:
+            #     annotator.delete_dataset(dataset_name=name)
+            # cli_utils.declare(f"Dataset '{name}' has now been deleted.")
+
+        @dataset.command(
+            "annotate", context_settings={"ignore_unknown_options": True}
+        )
+        @click.argument(
+            "name",
+            type=click.STRING,
+        )
+        @click.pass_obj
+        def dataset_annotate(annotator: "BaseAnnotator", name: str) -> None:
+            """Command to launch the annotation interface for a dataset.
+
+            Args:
+                annotator: The annotator stack component.
+                name: Name of the dataset
+            """
+            cli_utils.declare(
+                f"Launching the annotation interface for dataset '{name}'."
+            )
+            annotator.launch(url=annotator.get_annotation_url(name))
```

Comment:
Maybe a check here to see if a dataset with `name` exists before we try to launch it?

Reply:
Added